extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -195,14 +195,23 @@ func (a *genericActuator) Reconcile(ctx context.Context, worker *extensionsv1alp
 		}
 	}
 
-	if err := a.updateWorkerStatusMachineDeployments(ctx, worker, wantedMachineDeployments); err != nil {
+	if err := a.updateWorkerStatusMachineDeployments(ctx, worker, wantedMachineDeployments, false); err != nil {
 		return errors.Wrapf(err, "failed to update the machine deployments in worker status")
 	}
 
 	return nil
 }
 
-func (a *genericActuator) deployMachineDeployments(ctx context.Context, cluster *controller.Cluster, worker *extensionsv1alpha1.Worker, existingMachineDeployments *machinev1alpha1.MachineDeploymentList, wantedMachineDeployments worker.MachineDeployments, classKind string, clusterAutoscalerUsed bool) error {
+func (a *genericActuator) isRollingUpdateInProgress(wantedMachineDeployments extensionsworker.MachineDeployments, existingMachineClassNames sets.String) bool {
```

Comment:
Shouldn't this function capture these conditions? - https://github.com/gardener/gardener/pull/2459/files#diff-ce16933f7218046c410bbaed72f771b7R368.

Reply:
You are right. I did not want to change this check until I find a case, when comparing the MachineClasses is not sufficient. 
I found one: MachineClasses are not garuanteed to be garbage collected. So If there is already a MachineClass with the desired spec in the Cluster, this check does not recognize a RollingUpdate.

Reply:
The check for the rolling update needs therefore (as you suggested), to be the same as the check when the rolling update is complete. 
Hence I ll move the check to the function waitUntilWantedMachineDeploymentsAvailable()

Reply:
Yes, makes sense.

Reply:
Actually the check for a rolling update needs to enhanced with a check for valid machine sets.  The cache of the client requesting the machine deployments can be outdated.

Reply:
But this cache should be eventually consistent right? So at worse, it would only be a few seconds later, right?

Reply:
yes, but then the check that should wait for the rolling update to finish is already done. So it does not wait.
That leads to a Shoot beeing 100 % reconciled even though the rolling update is still in progress.